tests/integration/test_interactive.py

```diff
@@ -95,3 +118,110 @@ def test_interactive_submission_success(rf, local_repo, enable_network):
     with tempfile.TemporaryDirectory(suffix=f"repo-{ar_pk}") as path:
         git("clone", repo.url, path)
         local_run.main(path, ["run_all"])
+
+
+def test_interactive_publishing_report_success(client, release, slack_messages):
+    # set up the project…
+    project = ProjectFactory()
+    workspace = WorkspaceFactory(project=project, name=project.interactive_slug)
+
+    # … and the user
+    user = UserFactory()
+    ProjectMembershipFactory(project=project, user=user, roles=[InteractiveReporter])
+    client.force_login(user)
+
+    # "run" the job
+    job_request = JobRequestFactory(workspace=workspace)
+    JobFactory(job_request=job_request, status="succeeded")
+
+    report = ReportFactory(
+        release_file=release.files.first(), project=project, title="My report title"
+    )
+    analysis = AnalysisRequestFactory(
+        job_request=job_request, project=project, report=report
+    )
+
+    # this is a complex stack of factories so this is a little sense check to
+    # ensure the analysis is in the expected success state to allow us to
+    # request it's published
+    assert analysis.status == "succeeded"
+
+    # user can view their analysis page and the report is rendered there
+    with patch.object(
+        AnalysisRequestCreate, "get_opencodelists_api", new=FakeOpenCodelistsAPI
+    ):
+        response = client.get(project.get_interactive_url())
+        assert response.status_code == 200
+        # TODO: check rendered response has the report details
+        # assert report.title in response.rendered_content
+
+    # user creates a request to publish report
+    with patch.object(PublishRequestCreate, "get_github_api", new=FakeGitHubAPI):
+        response = client.post(analysis.get_publish_url(), follow=True)
+        assert response.status_code == 200
+        assert response.redirect_chain == [(analysis.get_absolute_url(), 302)]
+
+    # check ticket is requested
+    requests = PublishRequest.objects.filter(report=report)
+    assert requests.count() == 1
+
+    # grab the publish request to make some of the tests below a little easier
```

Comment:
YMMV, but I'm not sure this comment is needed because the code is clear enough without it, I think.